File: tests/fixtures/azure-blob-py/basic/no_imports.py
class FakeBlob:
    def download_blob(self): return None
    def upload_blob(self, data): return None
    def delete_blob(self): return None


class FakeContainer:
    def get_blob_client(self, key): return FakeBlob()


class FakeSvc:
    def get_container_client(self, name): return FakeContainer()


svc = FakeSvc()


def local_fetch():
    return svc.get_container_client("nope").get_blob_client("nope").download_blob()


def local_save():
    return svc.get_container_client("nope").get_blob_client("nope").upload_blob(b"x")

File: tests/fixtures/azure-blob-rs/basic/src/no_imports.rs
struct FakeBlob;
struct FakeContainer;
struct FakeSvc;

impl FakeBlob {
    pub async fn get(&self) {}
    pub async fn delete(&self) {}
}
impl FakeContainer {
    pub fn blob_client(&self, _k: &str) -> FakeBlob { FakeBlob }
}
impl FakeSvc {
    pub fn container_client(&self, _c: &str) -> FakeContainer { FakeContainer }
}

pub async fn local_fetch() {
    let svc = FakeSvc;
    svc.container_client("nope").blob_client("nope").get().await;
    svc.container_client("nope").blob_client("nope").delete().await;
}
